src/tools/tidy/src/ui_tests.rs

```diff
@@ -17,9 +17,19 @@ pub fn check(path: &Path, bad: &mut bool) {
                      &mut |_| false,
                      &mut |file_path| {
         if let Some(ext) = file_path.extension() {
-            if (ext == "stderr" || ext == "stdout") && !file_path.with_extension("rs").exists() {
-                println!("Stray file with UI testing output: {:?}", file_path);
-                *bad = true;
+            if ext == "stderr" || ext == "stdout" {
+                // Some files have the pattern TESTNAME.stderr, some have
+                // TESTNAME.REVISION.stderr.  Grab the TESTNAME from
+                // everything up to the first dot.
+                let testname = file_path.file_name().unwrap()
+                                        .to_str().unwrap()
+                                        .splitn(2, '.').next().unwrap();
+                if !file_path.with_file_name(testname)
+                             .with_extension("rs")
+                             .exists() {
+                    println!("Stray file with UI testing output: {:?}", file_path);
```

Comment:
cc @pnkfelix

Reply:
Grabbing everything up to first dot to derive the source file name sounds like a reasonable start to me.